db/error_handler.cc

```diff
@@ -412,6 +412,28 @@ void ErrorHandler::SetBGError(const Status& bg_status,
     recover_context_ = context;
     return;
   }
+  const bool wal_related_reason =
+      reason == BackgroundErrorReason::kWriteCallback ||
+      reason == BackgroundErrorReason::kMemTable ||
+      reason == BackgroundErrorReason::kFlush;
```

Comment:
This seems to be a larger set of error case than just WAL write errors?

Reply:
Yes, we don't have a reason specifically for WAL errors. These errors cover WAL write faults but could include more cases.

Reply:
Does that mean we stop auto-recovery for more write errors that we previously did auto recovery for? Let me think ...

Reply:
I'm worried we will now have way less auto recovery cases though it's only for temporary till we pursue the long-term bug fix. The document API is also very behind too https://github.com/facebook/rocksdb/wiki/Background-Error-Handling#auto-recovery-situations (so a doc day opportunity :)

How bad is it to add one more parameter SetBGError() to indicate whether it's a WAL write or not and note that it's a temporary solution. This is to avoid having a new error reason in the public API BackgroundErrorReason?

Reply:
Added a new parameter to `bool wal_related` to SetBGError() for this.